File: postgre/fsharp/create/postgre_create.fs
// -------------------------------------------------------------------
//
//	postgre_create.fs
//
//					Jul/16/2011
//
// -------------------------------------------------------------------
open System
open System.Data
open Npgsql

// -------------------------------------------------------------------
let table_drop_proc (conn: NpgsqlConnection) =
  printfn "*** table_drop_proc ***"
  let sql_str = "drop table cities"
  printfn "%s" sql_str
  let cmd = new NpgsqlCommand (sql_str, conn )
  let rowsAffected = cmd.ExecuteNonQuery()
  printfn "rowsAffected = %d" rowsAffected;;

// -------------------------------------------------------------------
let table_create_proc (conn: NpgsqlConnection) =
  printfn "*** table_create_proc ***"
  let sql_str = "create TABLE cities (" + "id varchar(10) NOT NULL PRIMARY KEY," + "name text," + "population int," + "date_mod text)"
  printfn "%s" sql_str
  let cmd = new NpgsqlCommand (sql_str, conn )
  let rowsAffected = cmd.ExecuteNonQuery()
  ();;

// -------------------------------------------------------------------
let data_row_proc (conn: NpgsqlConnection,id_a,name,population_a,str_date) =
  let fmt_aa = "insert into cities (id, Name, Population, date_mod) values ('"
  let fmt_bb = id_a + "','" + name + "'," + population_a.ToString ()
  let fmt_cc = ",'" + str_date + "')"
  let sql_str = fmt_aa + fmt_bb + fmt_cc
  let cmd = new NpgsqlCommand (sql_str, conn )
  let rowsAffected = cmd.ExecuteNonQuery()
  ();;

// -------------------------------------------------------------------
let data_insert_proc (conn: NpgsqlConnection) =
  printfn "*** data_insert_proc *** start ***"

  data_row_proc (conn,"t3461","広島",65215,"1997-7-21")
  data_row_proc (conn,"t3462","福山",76284,"1997-8-29")
  data_row_proc (conn,"t3463","東広島",14653,"1997-9-7")
  data_row_proc (conn,"t3464","呉",89176,"1997-2-19")
  data_row_proc (conn,"t3465","尾道",94521,"1997-11-2")
  data_row_proc (conn,"t3466","竹原",25248,"1997-10-7")
  data_row_proc (conn,"t3467","三次",19282,"1997-9-9")
  data_row_proc (conn,"t3468","大竹",12397,"1997-12-12")
  data_row_proc (conn,"t3469","府中",41725,"1997-5-23")

  ();;

// -------------------------------------------------------------------
[<EntryPoint>]
let main (args : string[] ) =
  printfn "*** 開始 ***"

  let str_connect = "Server=localhost;Port=5432;User Id=scott;Password=tiger;ENCODING=UNICODE;Database=city;"

  let conn = new NpgsqlConnection (str_connect)

  conn.Open ()

  table_drop_proc (conn)

  table_create_proc (conn)

  data_insert_proc (conn)

  conn.Close ()

  printfn "*** 終了 ***"

  0

// -------------------------------------------------------------------

File: postgre/csharp/create/postgre_create.cs
// -------------------------------------------------------------------
/*
	postgre_create.cs

					Sep/28/2011

*/
// -------------------------------------------------------------------
using	System;
using	System.Data;
using	System.Text;
using	Npgsql;

// -------------------------------------------------------------------
class postgre_create
{
// -------------------------------------------------------------------
static void Main (string[] args)
{
	Console.WriteLine ("*** 開始 ***");


	string server="localhost";
	string port="5432";
	string str_db="city";

	string str_connect = "Server=" + server + ";Port=" + port + 
	";User Id=scott;Password=tiger;" +
		"ENCODING=UNICODE;Database=" + str_db + ";";

	postgre_manipulate.table_drop_proc (str_connect);

	postgre_manipulate.table_create_proc (str_connect);

	DataTable dtable = table_manipulate.table_prepare_proc ();

	dtable = data_prepare_proc (str_connect,dtable);

	table_manipulate.display_proc (dtable);

	Console.WriteLine ("*** 終了 ***");
}

// -------------------------------------------------------------------
/* [6]: */
static DataTable data_prepare_proc
	 (string str_connect,DataTable dtable)
{
	dtable = data_row_proc
		(str_connect,dtable,"t3461",@"広島",25387,"2009-4-25");

	dtable = data_row_proc
		(str_connect,dtable,"t3462",@"福山",52146,"2009-7-9");

	dtable = data_row_proc
		(str_connect,dtable,"t3463",@"東広島",46291,"2009-9-19");

	dtable = data_row_proc
		(str_connect,dtable,"t3464",@"呉",75836,"2009-7-29");

	dtable = data_row_proc
		(str_connect,dtable,"t3465",@"尾道",46924,"2009-11-2");

	dtable = data_row_proc
		(str_connect,dtable,"t3466",@"竹原",71456,"2009-10-7");

	dtable = data_row_proc
		(str_connect,dtable,"t3467",@"三次",92391,"2009-9-9");

	dtable = data_row_proc
		(str_connect,dtable,"t3468",@"大竹",75843,"2009-8-16");

	dtable = data_row_proc
		(str_connect,dtable,"t3469",@"府中",71523,"2009-10-21");

	return	dtable;
}

// -------------------------------------------------------------------
/* [6-4]: */
static DataTable data_row_proc
	(string str_connect,DataTable dtable,
	string id_a,string name_a,int population_a,string str_date)
{
	DataRow drow = dtable.NewRow ();
	drow["ID"] = id_a;
	drow["NAME"] = name_a;
	drow["POPULATION"] = population_a.ToString ();
	drow["DATE_MOD"] = str_date;

	dtable.Rows.Add (drow);

	postgre_manipulate.insert_proc (str_connect,id_a,name_a,population_a,str_date);

	return	dtable;
}

// -------------------------------------------------------------------
}

// -------------------------------------------------------------------
